code/modules/mob/living/carbon/superior_animal/ai.dm

```diff
@@ -30,18 +22,16 @@
 
 	var/turf/our_turf = get_turf(src)
 	if (our_turf) //If we're not in anything, continue
-		for(var/mob/living/target_mob as anything in hearers(src, viewRange)) //as anything works because isvalid has a isliving check, change if necessary
-			if (isValidAttackTarget(target_mob))
+		for(var/mob/living/target_mob as anything in hearers(viewRange, src)) //as anything works because isvalid has a isliving check, change if necessary
+			if(!(target_mob?.faction == faction && !attack_same) && isValidAttackTarget(target_mob))
```

Comment:
Sanity check.
Don't bother checking for valid. Shaves off a single proc for hordes.
Could be done better, if the for() is rewritten.

Reply:
Most of the overhead from calling a proc is in the actual running of that proc. If that proc changes we want all the places that use that logic to update with it. This isn't so much an optimization as it is a gutting.